From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matteo Melli <matteom@ongres.com>
Date: Tue, 21 Apr 2026 00:00:00 +0000
Subject: [PATCH 9/9] fix: make gensql.pl path-independent

contrib/ivorysql_ora/gensql.pl, when invoked in "meson" mode, resolves
its inputs via CWD-relative paths ("../contrib/ivorysql_ora/..."). That
only works when the meson build directory sits one level under the
source root. rules_foreign_cc runs the build from an unrelated sandbox
dir, so perl fails with:

  Can't open perl script "../contrib/ivorysql_ora/gensql.pl":
  No such file or directory

Use FindBin::$Bin to anchor all lookups to the script's own directory,
and update meson.build to pass gensql.pl's absolute path via
meson.current_source_dir() so perl can locate the script itself.
---
 contrib/ivorysql_ora/gensql.pl   | 8 +++++---
 contrib/ivorysql_ora/meson.build | 2 +-
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/contrib/ivorysql_ora/gensql.pl b/contrib/ivorysql_ora/gensql.pl
--- a/contrib/ivorysql_ora/gensql.pl
+++ b/contrib/ivorysql_ora/gensql.pl
@@ -33,6 +33,8 @@
 use warnings FATAL => 'all';
 use Carp;
 
+use FindBin qw($Bin);
+
 use File::Spec;
 
 my @sql_set;
@@ -55,7 +57,7 @@ sub sql_merge
 	# Read information from ivorysql_ora_merge_sqls.
 	#     There are SQLs specified which need to be merged.
 	if ($first_arg eq 'meson') {
-		open(INFO, "<", File::Spec->rel2abs("../contrib/ivorysql_ora/ivorysql_ora_merge_sqls"))
+		open(INFO, "<", "$Bin/ivorysql_ora_merge_sqls")
 		|| croak "Could not open file ivorysql_ora_merge_sqls: $!";
 	}
 	else {
@@ -68,7 +70,7 @@ sub sql_merge
 		# Delete the last tailing "\n" of this line.
 		chomp($_);
 		if ($first_arg eq 'meson') {
-			push @sql_set, "../contrib/ivorysql_ora/$_";
+			push @sql_set, "$Bin/$_";
 		}
 		else
 		{
@@ -86,7 +88,7 @@ sub sql_merge
 
 		# Open(create if necessary) ivorysql_ora--x.x(--x.x).sql.
 		if ($first_arg eq 'meson'){
-			open(OUTFILE, ">contrib/ivorysql_ora/ivorysql_ora--$v.sql")
+			open(OUTFILE, ">", File::Spec->devnull())
                         || croak "Could not open ivorysql_ora--$v.sql : $!";
 		}
 		else {
diff --git a/contrib/ivorysql_ora/meson.build b/contrib/ivorysql_ora/meson.build
--- a/contrib/ivorysql_ora/meson.build
+++ b/contrib/ivorysql_ora/meson.build
@@ -36,7 +36,7 @@ endif
 
 custom_target('ivorysql_ora--1.0.sql',
   output: 'ivorysql_ora--1.0.sql',
-  command: [perl, '../contrib/ivorysql_ora/gensql.pl', 'meson', '1.0'],
+  command: [perl, meson.current_source_dir() / 'gensql.pl', 'meson', '1.0'],
   capture: true,
   install: true,
   install_dir: contrib_data_args['install_dir'],
-- 
2.39.5

